platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/resolveengine/oldresult/TransientConfigurationResultsBuilder.java

```diff
@@ -137,14 +138,10 @@ public void visitArtifacts(DependencyGraphNode from, LocalFileDependencyMetadata
     public TransientConfigurationResults load(final SelectedArtifactResults artifactResults) {
         synchronized (lock) {
             return cache.load(() -> {
-                try {
-                    return binaryData.read(decoder -> deserialize(decoder, artifactResults, buildOperationExecutor, resolutionHost));
-                } finally {
-                    try {
-                        binaryData.close();
-                    } catch (IOException e) {
-                        throw throwAsUncheckedException(e);
-                    }
+                try (BinaryData localBinaryData = binaryData) {
+                    return localBinaryData.read(decoder -> deserialize(decoder, artifactResults, buildOperationExecutor, resolutionHost));
+                } catch (IOException e) {
+                    throw throwAsUncheckedException(e); // todo why this double trouble?
```

Comment:
💅 Please remove this todo

Reply:
yes of course sorry, was just to check here again and in the code.